geospaas_processing/cli/download.py

```diff
@@ -0,0 +1,87 @@
+"""
+Script for cli of downloading repo. This script accepts the argunemt from sys.args by the
+help of python "argparse" and download the dataset files based on the input arguments.
+"""
+import argparse
+import os
+import subprocess
+import sys
+import time
+from datetime import datetime
+from distutils.util import strtobool
+import json
+import django
+import ast
+
+from dateutil.parser import parse
+from dateutil.relativedelta import relativedelta
+from dateutil.tz import tzutc
+from django.contrib.gis.geos import GEOSGeometry
+
+os.environ.setdefault('DJANGO_SETTINGS_MODULE', 'geospaas_processing.settings')
+django.setup()
+from geospaas.catalog.models import Dataset
+import geospaas_processing.downloaders as downloaders
+
+
+def main(arg):
+    cumulative_query = {}
+    if arg.query:
+        query_list = json.loads(arg.query)
+        for query in query_list:
+            cumulative_query.update(query)
```

Comment:
This is inconsistent with the help fo `-q` option. If your json is {"dataseturi__uri__contains": "osisaf"} then is serialised directly into dictionary and there is no need to loop over elements:
```suggestion
        cumulative_query = json.loads(arg.query)
```
In case of several queries, the value for input parameter should be:
'{"dataseturi__uri__contains": "osisaf", "source__instrument__short_name__icontains": "AMSR2"}'

Reply:
I have tested, what you have suggested, the way you create `cumulative_query` with that single line creates a list for `cumulative_query`. We need a dictionary insead of a list to work with the source code of downloading. It is because in the source code of downloading, we use `**` for filtering criteria and `**` is for a dictionary not a list.

Reply:
I agree, you need a dict. But this line does create dict:
```python
>>> json.loads('{"dataseturi__uri__contains": "osisaf", "source__instrument__short_name__icontains": "AMSR2"}')
{'dataseturi__uri__contains': 'osisaf', 'source__instrument__short_name__icontains': 'AMSR2'}
```

Reply:
Then I have to change the comment for correct explanation. Sorry for misleading help content. `safe_load` creates a list and json package just serialize and deserialize it.

Reply:
So the help content would be:
```python
"After deserialization, it must be a list of query that are readable by django filter."
    "for example a list of elements like ["dataseturi__uri__contains": "osisaf"]
```

Reply:
OK, so yaml.safe_load returns a list of dicts? Then help string should be `[{"dataseturi__uri__contains": "osisaf"}]`, right? But that makes API more complicated.  Can you simplify API to accept only a dict and then you convert the list to dict in the master script?

By the way. When I run script with `python download.py -q '{a="n"}'` it convert it to `args.query='{a=n}'` (information about quotes is lost). So I have to use backslash in the arguments: `python download.py -q '{a=\"n\"}'` to get correct json: `args.query='{a="n"}'`. Do you have the same? Can you mention that in the help string?

Reply:
Done in the next commit